pkg/detectors/gengo/gengo.go

```diff
@@ -29,8 +30,8 @@ var (
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
 	// Removed bounds since there are some cases where the start and end of the token is a special character
-	keyPat    = regexp.MustCompile(detectors.PrefixRegex([]string{"gengo"}) + `([ ]{0,1}[0-9a-zA-Z\[\]\-\(\)\{\}|_^@$=~]{64}[ \r\n]{1})`)
-	secretPat = regexp.MustCompile(detectors.PrefixRegex([]string{"gengo"}) + `([ ]{0,1}[0-9a-zA-Z\[\]\-\(\)\{\}|_^@$=~]{64}[ \r\n]{1})`)
+	keyPat    = regexp.MustCompile(detectors.PrefixRegex([]string{"gengo"}) + `([0-9a-zA-Z\[\]\-\(\)\{\}|_^@$=~]{64})\b`)
```

Comment:
For the reviewer: this token has a real nasty pattern to match. No `\b` in front of the capture group due to the fact that some of the token's alphabet _could_ be escaped by that in-front-of-capture-group `\b`. `[ ]{0,1}` removed as it didn't provide any value according to my tests (though please double check this work if you can). 

FWIW the resulting regex is: `(?i)(?:gengo)(?:.|[\n\r]){0,40}([0-9a-zA-Z\[\]\-\(\)\{\}|_^@$=~]{64})\b`

We should probably have a discussion on the fact that the keyPat and secretPat are the same regex. I'm sure there is a way to tighten something up here.

Reply:
Same issue as @rgmz pointed out in https://github.com/trufflesecurity/trufflehog/pull/2407#discussion_r1484813476